lib/Array.cry

```diff
@@ -3,10 +3,9 @@
  * Distributed under the terms of the BSD3 license (see LICENSE file)
  */
 
-/** The type and operations of the theory of arrays. */
-
 module Array where
 
+/** The type and operations of the theory of arrays. */
```

Comment:
This doesn't seem to make sense as documentation for the `Array` *type*. The doc-string for the `Array` type should explain what the type parameters are, and say something about what kind of values inhabit type `Array a b`.